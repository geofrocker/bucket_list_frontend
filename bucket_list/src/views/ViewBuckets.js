import React, {Component} from 'react';
import Header from '../components/Header';
import Sidebar from '../components/Sidebar';

class ViewBuckets extends Component{
    render(){
        return(
            <div>
                <Header/>
                <Sidebar/>
                <article className="content responsive-tables-page">
                    <div className="title-block">
                        <h1 className="title"> View Buckets </h1>
                    </div>

                    <section className="section">
                        <div className="row">
                            <div className="col-md-12">
                                <div className="card">
                                    <div className="card-block">
                                        <div className="card-title-block">
                                            <h3 className="title"> Your Buckets </h3>
                                        </div>
                                        <section className="example">
                                            <div className="table-flip-scroll">
                                                <table className="table table-striped table-bordered table-hover flip-content">
                                                    <thead className="flip-header">
                                                    <tr>
                                                        <th>Name</th>
                                                        <th>Description</th>
                                                        <th>Category</th>
                                                        <th>Date Created</th>
                                                        <th>Activities</th>
                                                        <th>Activities</th>
                                                        <th>Update</th>
                                                        <th>Action</th>
                                                    </tr>
                                                    </thead>
                                                    <tbody>
                                                    </tbody>
                                                </table>
                                            </div>
                                        </section>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </section>
                </article>
            </div>
        )
    }
}
export default ViewBuckets