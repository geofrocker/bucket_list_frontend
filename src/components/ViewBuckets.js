import React, {Component} from 'react';
import Header from './Header';
import Sidebar from './Sidebar';
import axios from 'axios';
import {Link, Redirect} from 'react-router-dom';


class ViewBuckets extends Component{
    /**
     * This component is used to view buckets.
     *  render --> Used to return jsx
     */

    constructor(props){
        super(props);
        this.state = {isAuthorized: false, data:[], login_redirect:false};
    }

    componentDidMount() {
        axios({
            url: 'https://ridge-bucket-list-api.herokuapp.com/api/v1/bucketlists/',
            method: "GET",
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }
        })
            .then((response)=>{
                this.setState({data:response.data.buckets, isAuthorized: true});
            })

            .catch((xhr) =>{
                this.setState({login_redirect:true})

            });
    }

    render(){
        if(this.state.login_redirect){
            return(
                <Redirect to={'/login/'}/>
            )
        }

        else if (!this.state.isAuthorized){
            return(
                <div>
                    <article className="content item-editor-page">

                    </article>
                </div>
            )
        }
        if(this.state.login_redirect){
            return(
                <Redirect to={'/login/'}/>
            )
        }

        return(
            <div>
                <Header/>
                <Sidebar/>
                <article className="content responsive-tables-page">
                    <div className="title-block">
                        <h1 className="title"> View Buckets </h1>
                    </div>

                    <section className="section">
                        <div className="row">
                            <div className="col-md-12">
                                <div className="card">
                                    <div className="card-block">
                                        <div className="card-title-block">
                                            <h3 className="title"> Your Buckets </h3>
                                        </div>
                                        <section className="example">
                                            <div className="table-flip-scroll">
                                                <table id="buckets_table" className="table table-striped table-bordered table-hover flip-content">
                                                    <thead className="flip-header">
                                                    <tr>
                                                        <th>Name</th>
                                                        <th>Description</th>
                                                        <th>Date Created</th>
                                                        <th>Activities</th>
                                                        <th>Activities</th>
                                                        <th>Action</th>
                                                    </tr>
                                                    </thead>
                                                    <tbody>{this.state.data.map(function(item, key){
                                                        return (
                                                            <tr key={key}>
                                                                <Link to={'/bucketlists/update/' + item.id}><td>{item.bucket_name}</td></Link>
                                                                <td>{item.description}</td>
                                                                <td>{item.created}</td>
                                                                <td className="center">
                                                                    <button type="button" className="btn btn-success-outline btn-sm">
                                                                        <Link to={'/bucketlists/' + item.id + '/items/create/'}><td>Add</td></Link>
                                                                    </button>
                                                                </td>
                                                                <td className="center" >
                                                                    <button type="button" className="btn btn-success-outline btn-sm">
                                                                        <Link to={'/bucketlists/' + item.id + '/items/view/'}><td>View</td></Link>
                                                                    </button>
                                                                </td>
                                                                <td className="center" >
                                                                    <button type="button" className="btn btn-danger-outline btn-sm">
                                                                        <Link to={'/bucketlists/delete/' + item.id}><td>Delete</td></Link>
                                                                    </button>
                                                                </td>
                                                            </tr>
                                                        )
                                                    })}
                                                    </tbody>
                                                </table>
                                            </div>
                                        </section>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </section>
                </article>
            </div>
        )
    }
}

export default ViewBuckets;
