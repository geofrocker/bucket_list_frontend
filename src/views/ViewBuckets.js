import React, {Component} from 'react';
import Header from '../components/Header';
import Sidebar from '../components/Sidebar';
import axios from 'axios';
import {Link} from 'react-router-dom';


class ViewBuckets extends Component{

    constructor(props){
        super(props);
        this.state = {isAuthorized: false, data:[]};
        this.handleUpdate = this.handleUpdate.bind(this)
    }

    handleUpdate(){
        console.log('awesome')
    }

    componentDidMount() {
        axios({
            url: 'http://127.0.0.1:5000/api/v1/bucketlists/',
            method: "GET",
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }
        })
            .then((response)=>{
                this.setState({data:response.data.buckets, isAuthorized: true});

            })
            .catch((xhr) =>{
                window.localStorage.setItem('isLoggedIn', false)

            });
    }

    render(){
        if (!this.state.isAuthorized){
            return(
                <div>
                    <article className="content item-editor-page">
                        <div className="card card-block">
                            <p>Unauthorized! Please <Link to={'/login/'}>Login</Link></p>
                        </div>
                    </article>
                </div>
            )
        }

        return(
            <div>
                <Header/>
                <Sidebar/>
                <article className="content responsive-tables-page">
                    <div className="title-block">
                        <h1 className="title"> View Buckets </h1>
                    </div>

                    <section className="section">
                        <div className="row">
                            <div className="col-md-12">
                                <div className="card">
                                    <div className="card-block">
                                        <div className="card-title-block">
                                            <h3 className="title"> Your Buckets </h3>
                                        </div>
                                        <section className="example">
                                            <div className="table-flip-scroll">
                                                <table className="table table-striped table-bordered table-hover flip-content">
                                                    <thead className="flip-header">
                                                    <tr>
                                                        <th>Name</th>
                                                        <th>Description</th>
                                                        <th>Date Created</th>
                                                        <th>Activities</th>
                                                        <th>Activities</th>
                                                        <th>Action</th>
                                                    </tr>
                                                    </thead>
                                                    <tbody>{this.state.data.map(function(item, key){
                                                        return (
                                                            <tr key={key}>
                                                                <Link to={'/bucketlists/update/' + item.id}><td>{item.bucket_name}</td></Link>
                                                                <td>{item.description}</td>
                                                                <td>{item.created}</td>
                                                                <td className="center">
                                                                    <button type="button" className="btn btn-success-outline btn-sm">
                                                                        <Link to={'/bucketlists/' + item.id + '/items/create/'}><td>Add</td></Link>
                                                                    </button>
                                                                </td>
                                                                <td className="center" >
                                                                    <button type="button" className="btn btn-success-outline btn-sm">
                                                                        <Link to={'/bucketlists/' + item.id + '/items/view/'}><td>View</td></Link>
                                                                    </button>
                                                                </td>
                                                                <td className="center" >
                                                                    <button type="button" className="btn btn-danger-outline btn-sm">
                                                                        <Link to={'/bucketlists/delete/' + item.id}><td>Delete</td></Link>
                                                                    </button>
                                                                </td>
                                                            </tr>
                                                        )
                                                    })}
                                                    </tbody>
                                                </table>
                                            </div>
                                        </section>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </section>
                </article>
            </div>
        )
    }
}

export default ViewBuckets;
