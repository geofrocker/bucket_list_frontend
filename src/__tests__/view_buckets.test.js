import React from 'react';
import ViewBuckets from '../views/ViewBuckets';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import {MemoryRouter} from 'react-router-dom';

describe('<ViewBuckets />', ()=>{
    beforeEach(() => {
        global.localStorage = {
            token: 'someToken',
            getItem: function () {
                return 'someToken'
            }
        };

    });

    it('contains bucket table', function () {
        const wrapper = mount(
            <MemoryRouter>
                <ViewBuckets/>
            </MemoryRouter>);

        wrapper.setState({data:[{bucket_name:'test_bucket', description:"test description"}], isAuthorized:true});

        setTimeout(() => {
            expect(wrapper.find("#buckets_table").length === 1);

        }, 60);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

});

