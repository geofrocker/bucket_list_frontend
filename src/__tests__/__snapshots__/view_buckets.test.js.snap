// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewBuckets /> contains bucket table 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "mnkpnd",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "mnkpnd",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ViewBuckets>
      <div>
        <article
          className="content item-editor-page"
        />
      </div>
    </ViewBuckets>
  </Router>
</MemoryRouter>
`;
